actors/market/src/lib.rs

```diff
@@ -846,66 +869,45 @@ impl Actor {
 
         rt.transaction(|st: &mut State, rt| {
             let last_cron = st.last_cron;
-            let mut provider_deals_to_remove: BTreeMap<
-                ActorID,
-                BTreeMap<SectorNumber, Vec<DealID>>,
-            > = BTreeMap::new();
+            let mut provider_deals_to_remove =
+                BTreeMap::<ActorID, BTreeMap<SectorNumber, Vec<DealID>>>::new();
             let mut new_updates_scheduled: BTreeMap<ChainEpoch, Vec<DealID>> = BTreeMap::new();
             let mut epochs_completed: Vec<ChainEpoch> = vec![];
 
             for i in (last_cron + 1)..=rt.curr_epoch() {
                 let deal_ids = st.get_deals_for_epoch(rt.store(), i)?;
 
                 for deal_id in deal_ids {
-                    let deal = st.get_proposal(rt.store(), deal_id)?;
-                    let dcid = deal_cid(rt, &deal)?;
-                    let state = st.find_deal_state(rt.store(), deal_id)?;
+                    let deal_proposal = match st.find_proposal(rt.store(), deal_id)? {
+                        Some(dp) => dp,
+                        // proposal might have been cleaned up by manual settlement or termination prior to reaching
+                        // this scheduled cron tick. nothing more to do for this deal
+                        None => continue,
+                    };
```

Comment:
```suggestion
                    // proposal might have been cleaned up by manual settlement or termination prior to reaching
                    // this scheduled cron tick. nothing more to do for this deal
                    let Some(deal_proposal) = st.find_proposal(rt.store(), deal_id)? else { continue };
```

I have a goal to make you say "I hate rust" at least once per PR (feel free to ignore this suggestion).

Reply:
Thanks!